File: src/main/java/io/github/gdpl2112/model/StorageInfo.java
package io.github.gdpl2112.model;

import lombok.Data;

import java.text.DecimalFormat;

@Data
public class StorageInfo {
    private long limit;           // 存储限制（字节）
    private long used;            // 已使用存储（字节）
    private long remaining;        // 剩余存储（字节）
    private double percentage;     // 使用百分比
    private String limitFormatted;    // 格式化的存储限制
    private String usedFormatted;     // 格式化的已使用存储
    private String remainingFormatted; // 格式化的剩余存储


    /**
     * 格式化文件大小
     * @param bytes 字节数
     * @return 格式化后的文件大小字符串
     */
    public static String formatFileSize(long bytes) {
        if (bytes <= 0) return "0 Bytes";

        final String[] units = new String[]{"Bytes", "KB", "MB", "GB", "TB"};
        int digitGroups = (int) (Math.log10(bytes) / Math.log10(1024));
        return new DecimalFormat("#,##0.#").format(bytes / Math.pow(1024, digitGroups)) + " " + units[digitGroups];
    }
}

File: gofiles/model/StorageInfo.go
package model

import (
	"fmt"
	"math"
)

// StorageInfo 存储信息模型
type StorageInfo struct {
	Limit              int64   `json:"limit"`
	Used               int64   `json:"used"`
	Remaining          int64   `json:"remaining"`
	Percentage         float64 `json:"percentage"`
	LimitFormatted     string  `json:"limitFormatted"`
	UsedFormatted      string  `json:"usedFormatted"`
	RemainingFormatted string  `json:"remainingFormatted"`
}

// FormatFileSize 格式化文件大小
func FormatFileSize(bytes int64) string {
	if bytes <= 0 {
		return "0 Bytes"
	}

	units := []string{"Bytes", "KB", "MB", "GB", "TB"}
	digitGroups := int(math.Log10(float64(bytes)) / math.Log10(1024))
	if digitGroups >= len(units) {
		digitGroups = len(units) - 1
	}

	return fmt.Sprintf("%.1f %s", float64(bytes)/math.Pow(1024, float64(digitGroups)), units[digitGroups])
}
